src/main.rs

```diff
@@ -77,7 +88,7 @@ fn main() {
     // a palette of hex strings which are coloured to match.
     // See https://alexwlchan.net/2021/04/coloured-squares/
     for c in rgb {
-        let display_value = format!("#{:02x}{:02x}{:02x}", c.red, c.green, c.blue);
+        let display_value = colour_output.format_colour(c);
```

Comment:
It feels a bit weird to me that we're acting on `colour_output`, not on `c`. Is there any way we could rejig the `impl`s so this can become

```suggestion
        let display_value = c.notate_as(notation);
```

Reply:
Also the comment directly above this `for` loop will need updating, because it's no longer a palette of hex strings.